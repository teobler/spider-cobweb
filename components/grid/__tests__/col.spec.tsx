import React from 'react';
import renderer from 'react-test-renderer';
import Col from '../col';
import Row from '../row';

describe('<Col />', () => {
  it('should render a full col in row', () => {
    const layoutJson = renderer
      .create(
        <Row>
          <Col span={24} />
        </Row>,
      )
      .toJSON();
    expect(layoutJson).toMatchSnapshot();
  });

  it('should render a col with offset in row', () => {
    const layoutJson = renderer
      .create(
        <Row>
          <Col span={12} offset={12} />
        </Row>,
      )
      .toJSON();
    expect(layoutJson).toMatchSnapshot();
  });

  it('should render two col with gutter box', () => {
    const layoutJson = renderer
      .create(
        <Row gutter={16}>
          <Col span={12}>test</Col>
          <Col span={12}>test</Col>
        </Row>,
      )
      .toJSON();
    expect(layoutJson).toMatchSnapshot();
  });

  it('should render two adaptive col with offset', () => {
    const layoutJson = renderer
      .create(
        <Row>
          <Col
            span={1}
            xs={{ span: 24, offset: 1 }}
            sm={{ span: 12, offset: 2 }}
            md={{ span: 10, offset: 3 }}
            lg={{ span: 8, offset: 4 }}
            xl={{ span: 6, offset: 5 }}
            className="spider-col-demo"
          >
            col-test
          </Col>
          <Col
            span={1}
            xs={{ span: 24, offset: 1 }}
            sm={{ span: 12, offset: 2 }}
            md={{ span: 10, offset: 3 }}
            lg={{ span: 8, offset: 4 }}
            xl={{ span: 6, offset: 5 }}
            className="spider-col-demo"
          >
            col-test
          </Col>
        </Row>,
      )
      .toJSON();
    expect(layoutJson).toMatchSnapshot();
  });
});
