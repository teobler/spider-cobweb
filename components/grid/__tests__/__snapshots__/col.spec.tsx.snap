// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col /> should render a col with offset in row 1`] = `
<div
  className="spider-row"
  style={Object {}}
>
  <div
    className="spider-col-12 spider-col-offset-12"
    style={Object {}}
  />
</div>
`;

exports[`<Col /> should render a full col in row 1`] = `
<div
  className="spider-row"
  style={Object {}}
>
  <div
    className="spider-col-24"
    style={Object {}}
  />
</div>
`;

exports[`<Col /> should render two adaptive col with offset 1`] = `
<div
  className="spider-row"
  style={Object {}}
>
  <div
    className="spider-col-1 spider-xs-col-24 spider-xs-col-offset-1 spider-sm-col-12 spider-sm-col-offset-2 spider-md-col-10 spider-md-col-offset-3 spider-lg-col-8 spider-lg-col-offset-4 spider-xl-col-6 spider-xl-col-offset-5 spider-col-demo"
    lg={
      Object {
        "offset": 4,
        "span": 8,
      }
    }
    md={
      Object {
        "offset": 3,
        "span": 10,
      }
    }
    sm={
      Object {
        "offset": 2,
        "span": 12,
      }
    }
    style={Object {}}
    xl={
      Object {
        "offset": 5,
        "span": 6,
      }
    }
    xs={
      Object {
        "offset": 1,
        "span": 24,
      }
    }
  >
    col-test
  </div>
  <div
    className="spider-col-1 spider-xs-col-24 spider-xs-col-offset-1 spider-sm-col-12 spider-sm-col-offset-2 spider-md-col-10 spider-md-col-offset-3 spider-lg-col-8 spider-lg-col-offset-4 spider-xl-col-6 spider-xl-col-offset-5 spider-col-demo"
    lg={
      Object {
        "offset": 4,
        "span": 8,
      }
    }
    md={
      Object {
        "offset": 3,
        "span": 10,
      }
    }
    sm={
      Object {
        "offset": 2,
        "span": 12,
      }
    }
    style={Object {}}
    xl={
      Object {
        "offset": 5,
        "span": 6,
      }
    }
    xs={
      Object {
        "offset": 1,
        "span": 24,
      }
    }
  >
    col-test
  </div>
</div>
`;

exports[`<Col /> should render two col with gutter box 1`] = `
<div
  className="spider-row"
  style={
    Object {
      "margin": "0 -8px",
    }
  }
>
  <div
    className="spider-col-12"
    style={
      Object {
        "padding": "0 8px",
      }
    }
  >
    <div
      className="gutter-box"
    >
      test
    </div>
  </div>
  <div
    className="spider-col-12"
    style={
      Object {
        "padding": "0 8px",
      }
    }
  >
    <div
      className="gutter-box"
    >
      test
    </div>
  </div>
</div>
`;
